modules/core-components/src/main/java/org/mule/runtime/core/internal/routing/forkjoin/AbstractForkJoinStrategyFactory.java

```diff
@@ -130,6 +133,25 @@ public ForkJoinStrategy createForkJoinStrategy(ProcessingStrategy processingStra
     };
   }
 
+  private void handleTimeoutExceptionIfPresent(Scheduler timeoutScheduler,
+                                               List<Pair<CoreEvent, EventProcessingException>> listBooleanPair) {
+    listBooleanPair
+        .forEach(
+                 pair -> {
+                   final Optional<Error> error = pair.getFirst().getError();
+                   if (error.isPresent() &&
+                       error.get().getCause() instanceof TimeoutException &&
+                       error.get().getCause().getMessage()
+                           .contains(TIMEOUT_EXCEPTION_DETAILED_DESCRIPTION_PREFIX)) {
+                     EventContext context = pair.getFirst().getContext();
+                     if (context instanceof AbstractEventContext) {
+                       ((AbstractEventContext) context).forEachChild(ctx -> timeoutScheduler
+                           .submit(() -> ctx.error(new MessagingException(pair.getFirst(), error.get().getCause()))));
+                     }
+                   }
```

Comment:
What about submitting just one task for all the child contexts?
Something like:
```suggestion
                     EventContext context = pair.getFirst().getContext();
                     if (context instanceof AbstractEventContext) {
                       timeoutScheduler
                           .submit(() -> ((AbstractEventContext) context).forEachChild(ctx -> ctx.error(new MessagingException(pair.getFirst(), error.get().getCause()))));
                     }
                   }
```

Reply:
I would really like to understand why is this happening. How come scheduling 70 different tasks is working but scheduling just one is not... What is it that we don't know?

Reply:
If we're purely considering the resources used probably nesting of SG routes that I've created may have caused recursive calls of error(...) which also deals with reentrant locks. Not sure if there if anything there, but throwing ideas around.